python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/shape_lib_gen.py

```diff
@@ -919,6 +919,13 @@ def aten〇squeeze〇dim(self: List[int], dim: int) -> List[int]:
 def prim〇NumToTensor〇Scalar(a: float) -> List[int]:
     return []
 
+def prims〇squeeze(a: List[int], dimensions: List[int]) -> List[int]:
+    out = a
+    num_dims = len(dimensions)
+    for i in range(num_dims):
+        out = upstream_shape_functions.squeeze(out, dimensions[num_dims - i - 1])
```

Comment:
You need to sort `dimensions` first, right? You will probably need to add the canonicalization pattern for `sort` from you other PR here: https://github.com/llvm/torch-mlir/blob/1d949f3ac27e89232ec376fe2d3b2ea6b5c33674/lib/Dialect/Torch/Transforms/SimplifyShapeCalculations.cpp#L421

You also need to handle duplicate dimensions. I think you can do this by keeping track of the dimension used in the previous iteration, then checking if it equals the new dimension (after sorting, of course)

Reply:
Actually, duplicate dimensions are allowed for this op.

Reply:
Right, but duplicate dimensions need to only be squeezed once. In other words, you need to remove the duplicates.

```python
In [12]: a = torch.rand((1,2,1,4,5,6))

In [13]: torch.ops.prims.squeeze(a, [0,2,2,0,0]).size()
Out[13]: torch.Size([2, 4, 5, 6])
```

Also, can you add an e2e test that checks this behavior and the dimensions out of order behavior?